appIcons.js

```diff
@@ -689,11 +695,14 @@ const MyAppIcon = new Lang.Class({
 
         let rgb = colorUtils.HSVtoRGB(hsv.h, hsv.s, hsv.v);
 
-        return {
+        // Cache the result.
+        this._backgroundColors = {
```

Comment:
I was trying to figure out why the caching was not working... It turns out the variable here ends with an "s". So the color is actually never cached!

```
this._backgroundColor = null;
this._backgroundColors = {
```